src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -2611,7 +2615,13 @@ private String getSignatureForStaticTTEFn(String propName, FunctionType ftype) {
     private String getPromiseMethod(String propName, String className) {
       switch (propName) {
         case "resolve":
-          return "resolve < T >(value: " + className + " < T > | T): " + className + " < T >;";
+          // TODO(lucassloan): goog.Promise has bad types that are coerced to any, so explicitly emit any
+          // and fix when the callers have been fixed.
+          if (className.equals("ಠ_ಠ.clutz.goog.Promise")) {
+            return "resolve < T >(value: " + className + " < T , any > | T): any;";
```

Comment:
OOC, what happens if you do `: <T, any>` for the return type too?

Reply:
That's the correct type, and all the bad user's code is surfaced.